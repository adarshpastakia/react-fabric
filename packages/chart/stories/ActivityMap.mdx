import { Collapsable } from "@react-fabric/core";
import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { ActivityMap } from "../src";
import * as ActivityMapStories from "./ActivityMap.stories";

<Meta of={ActivityMapStories} />

<Title>Activity Map</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Activity Map</Subheading>
  </div>
</Anchor>

<Description of={ActivityMap} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={ActivityMapStories.Example} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="data">
  <Subheading>Data Format</Subheading>
</Anchor>

```ts
// Hours of day
{
    time: "day-hour",
    data:[
        // day 1
        [[hour, day, count],...],
        // day 2
        [[hour, day, count],...]
    ]
}

// Days of month
{
    time: "month-day",
    data:[
        // month 1
        [[day, month, count],...],
        // month 2
        [[day, month, count],...]
    ]
}
```
