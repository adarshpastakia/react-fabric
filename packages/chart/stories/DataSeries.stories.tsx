/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { faker } from "@faker-js/faker";
import { Button, Panel } from "@react-fabric/core";
import type { Meta, StoryObj } from "@storybook/react";
import { useCallback, useEffect, useState } from "react";
import { DataSeries } from "../src";

const meta: Meta<typeof DataSeries> = {
  component: DataSeries,
  title: "@charts/DataSeries",
  parameters: {
    layout: "centered",
    controls: { exclude: "children" },
  },
};

export default meta;
type Story = StoryObj<typeof DataSeries>;

export const Example: Story = {
  render: (args) => {
    const [data, setData] = useState<AnyObject>({});
    const loadData = useCallback(() => {
      const categories = ["Jan", "Feb", "Mar", "Apr", "May", "Jun"];
      setData({
        categoryAxisName: "Months",
        valueAxisName: "Items",
        categories,
        data: [
          {
            id: faker.string.alpha(5),
            label: faker.animal.dog(),
            values: categories.map(() =>
              faker.number.int({ min: 100, max: 500 }),
            ),
          },
          {
            id: faker.string.alpha(5),
            label: faker.animal.dog(),
            values: categories.map(() =>
              faker.number.int({ min: 100, max: 500 }),
            ),
          },
          {
            id: faker.string.alpha(5),
            label: faker.animal.dog(),
            values: categories.map(() =>
              faker.number.int({ min: 100, max: 500 }),
            ),
          },
        ],
      });
    }, []);
    useEffect(() => {
      loadData();
    }, []);
    return (
      <Panel
        width="48rem"
        height="24rem"
        title="Data Series chart"
        expandable
        actions={
          <Button
            aria-label="loadData"
            variant="link"
            icon="mdi mdi-refresh"
            onClick={loadData}
          />
        }
      >
        <DataSeries {...args} {...data} />
      </Panel>
    );
  },
  args: {},
};
