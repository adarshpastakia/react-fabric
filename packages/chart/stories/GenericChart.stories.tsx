/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { Button, Panel, ToggleButtonGroup } from "@react-fabric/core";
import type { Meta, StoryObj } from "@storybook/react";
import React, { useEffect, useState } from "react";
import { GenericChart } from "../src";
import coffeeJson from "./coffee";
import sankeyJson from "./sankey";

const meta: Meta<typeof GenericChart> = {
  component: GenericChart,
  title: "@charts/GenericChart",
  parameters: {
    layout: "centered",
    controls: { exclude: "children" },
  },
};

export default meta;
type Story = StoryObj<typeof GenericChart>;

export const TreeMap: Story = {
  render: (args) => {
    const [type, setType] = useState("treemap");
    const [series, setSeries] = useState<AnyObject>({});

    useEffect(() => {
      if (type === "treemap") {
        setSeries({
          id: "coffee-list",
          name: "Coffee list",
          type: "treemap",
          data: coffeeJson,
          universalTransition: true,
          leafDepth: 2,
          upperLabel: {
            show: true,
            height: 30,
          },
          levels: [
            {
              itemStyle: {
                borderColor: "#777",
                borderWidth: 0,
                gapWidth: 1,
              },
              upperLabel: {
                show: false,
              },
            },
            {
              colorSaturation: [0.35, 0.5],
              itemStyle: {
                borderWidth: 5,
                gapWidth: 1,
                borderColorSaturation: 0.6,
              },
            },
          ],
        });
      }
      if (type === "sunburst") {
        setSeries({
          id: "coffee-list",
          name: "Coffee list",
          type: "sunburst",
          radius: ["20%", "90%"],
          data: coffeeJson,
          universalTransition: true,
          levels: [
            {},
            {
              label: {
                rotate: "tangential",
              },
            },
            {
              r0: "35%",
              r: "70%",
              label: {
                align: "right",
              },
            },
            {
              label: {
                position: "outside",
              },
            },
          ],
        });
      }
    }, [type]);
    return (
      <Panel width="48rem" height="24rem" title="Generic chart" expandable>
        <GenericChart
          {...args}
          series={[series]}
          legend={{ show: false }}
          actions={
            <ToggleButtonGroup value={type} onChange={setType}>
              <Button
                value="treemap"
                icon="mdi mdi-view-dashboard-variant"
                aria-label="tree"
              />
              <Button
                value="sunburst"
                icon="mdi mdi-circle-double"
                aria-label="burst"
              />
            </ToggleButtonGroup>
          }
        />
      </Panel>
    );
  },
  args: {},
};

export const Sankey: Story = {
  render: (args) => {
    const [series, setSeries] = useState<AnyObject>({});

    useEffect(() => {
      setSeries({
        type: "sankey",
        emphasis: {
          focus: "adjacency",
        },
        nodeAlign: "right",
        data: sankeyJson.nodes,
        links: sankeyJson.links,
        lineStyle: {
          color: "source",
          curveness: 0.5,
        },
      });
    }, []);
    return (
      <Panel width="48rem" height="24rem" title="Generic chart" expandable>
        <GenericChart {...args} series={[series]} legend={{ show: false }} />
      </Panel>
    );
  },
  args: {},
};
