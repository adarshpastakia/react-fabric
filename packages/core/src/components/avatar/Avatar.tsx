/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { isSvgPath } from "@react-fabric/utilities";
import BoringAvatar from "boring-avatars";
import classNames from "classnames";
import { useState } from "react";
import { useMemoDebugger } from "../../hooks/useEffectDebugger";
import {
  type AriaProps,
  type ColorType,
  type CssProp,
  type RefProp,
  type SizeType,
  type TestProps,
} from "../../types";
import { getBgClass, getColor, getColorClass } from "../../utils";
import classes from "./Avatar.module.css";

export interface AvatarProps extends CssProp, AriaProps, TestProps, RefProp {
  /**
   * avatar name for generated fallback
   */
  name: string;
  /**
   * svg path
   */
  fallbackIcon?: string;
  /**
   * avatar image url
   */
  avatar?: string;
  /**
   * avatar variant
   */
  variant?: "text" | "beam" | "pixel" | "bauhaus";
  /**
   * background color (CSS color / tailwind color)
   */
  bg?: ColorType | string;
  /**
   * icon color (CSS color / tailwind color)
   */
  color?: ColorType | string;
  /**
   * icon size
   */
  size?: SizeType | number | string;
  /**
   * viewbox for svg path (default 24)
   */
  viewBox?: string;
  /**
   * rounded icon
   */
  rounded?: boolean;
  /**
   * click handler
   */
  onClick?: React.MouseEventHandler;
}

const SizeMap: KeyValue<string> = {
  xs: "0.875rem",
  sm: "1.125rem",
  md: "1.5rem",
  lg: "2rem",
  xl: "2.5rem",
};

/**
 * An avatar is a graphical representation of a user's identity.
 */
export const Avatar = ({
  ref,
  name,
  avatar,
  fallbackIcon,
  bg,
  color,
  size = "",
  rounded,
  variant = "beam",
  className,
  viewBox = "0 0 24 24",
  onClick,
  ...aria
}: AvatarProps) => {
  const [fallback, setFallback] = useState(true);

  /** ***************** style map *******************/
  const styles = useMemoDebugger(
    () => {
      const s: KeyValue = {};
      if (bg) {
        s.backgroundColor = getColor(bg);
      }
      if (color) {
        s.color = getColor(color);
      }
      if (size && !(size in SizeMap)) {
        s.fontSize = size;
      }
      if (size && size in SizeMap) {
        s.fontSize = SizeMap[size];
      }
      return s;
    },
    [bg, color, size],
    "Avatar styles",
  );

  /** ***************** render icon *******************/
  const avatarImage = useMemoDebugger(
    () => {
      setFallback(!avatar);
      return (
        avatar && (
          <div className="contents">
            <div
              className={classNames(
                classes.imgPlaceholder,
                "absolute inset-0 bg-gray animate-pulse",
              )}
            />
            <img
              src={avatar}
              alt={avatar}
              className="align-baseline"
              onError={() => setFallback(true)}
              onLoad={(e) => e.currentTarget.previousElementSibling?.remove()}
            />
          </div>
        )
      );
    },
    [avatar],
    "Avatar icon",
  );

  const fallbackAvatar = useMemoDebugger(
    () => {
      if (fallbackIcon) {
        return isSvgPath(fallbackIcon) ? (
          <svg viewBox={viewBox}>
            <path fill="currentColor" d={fallbackIcon.toString()} />
          </svg>
        ) : (
          <i className={fallbackIcon} />
        );
      }
      if (variant === "text") {
        const [, first, second, group] =
          name.match(/(?=(\w).* (\w).*)|(?=(\w\w).*)/) ?? [];
        return (
          <svg role="img">
            <text
              x="50%"
              y="50%"
              dy=".1em"
              dominantBaseline="middle"
              textAnchor="middle"
              style={{ fontSize: ".5em", fontWeight: 500 }}
            >
              {group ?? `${first}${second}`}
            </text>
          </svg>
        );
      }
      return <BoringAvatar square size={64} variant={variant} name={name} />;
    },
    [name, variant, fallbackIcon],
    "Avatar fallback",
  );

  return (
    <dfn
      role="presentation"
      className={classNames(
        classes.avatar,
        className,
        "select-none relative overflow-hidden inline-block box-content leading-none not-italic text-center",
        bg && getBgClass(bg),
        color && getColorClass(color),
        !rounded && "rounded",
        rounded && "rounded-full",
      )}
      onClick={onClick}
      style={styles}
      data-clickable={!!onClick}
      ref={ref}
      {...aria}
    >
      {!fallback && avatarImage}
      {fallback && fallbackAvatar}
    </dfn>
  );
};
