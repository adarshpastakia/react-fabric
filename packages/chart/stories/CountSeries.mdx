import { Collapsable } from "@react-fabric/core";
import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { CountSeries } from "../src";
import * as CountSeriesStories from "./CountSeries.stories";

<Meta of={CountSeriesStories} />

<Title>Count Series</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Count Series</Subheading>
  </div>
</Anchor>

<Description of={CountSeries} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={CountSeriesStories.Example} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="data">
  <Subheading>Data Format</Subheading>
</Anchor>

```ts
// Single list of count values
[{ id, label, count },...];
```
