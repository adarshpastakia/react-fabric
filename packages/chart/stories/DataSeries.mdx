import { Collapsable } from "@react-fabric/core";
import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { DataSeries } from "../src";
import * as DataSeriesStories from "./DataSeries.stories";

<Meta of={DataSeriesStories} />

<Title>Data Series</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Data Series</Subheading>
  </div>
</Anchor>

<Description of={DataSeries} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={DataSeriesStories.Example} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="data">
  <Subheading>Data Format</Subheading>
</Anchor>

```ts
// List of data series
[{ id, label, values: [...] },...];
```
