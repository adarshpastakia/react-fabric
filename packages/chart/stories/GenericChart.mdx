import { Collapsable } from "@react-fabric/core";
import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { GenericChart } from "../src";
import * as GenericChartStories from "./GenericChart.stories";

<Meta of={GenericChartStories} />

<Title>Generic Chart</Title>

<Anchor storyId="animations">
  <div className="hidden-anchor">
    <Subheading>Tree Map / Sunburst</Subheading>
  </div>
</Anchor>

<Description of={GenericChart} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={GenericChartStories.TreeMap} />
</Collapsable>

<Anchor storyId="sankey">
  <Subheading>Sankey</Subheading>
</Anchor>
<Description of={GenericChartStories.Sankey} />
<Canvas withToolbar of={GenericChartStories.Sankey} />
<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={GenericChartStories.Sankey} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />
