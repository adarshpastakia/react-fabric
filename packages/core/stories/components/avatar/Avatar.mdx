import {
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { Fragment } from "react";
import { Avatar, AvatarGroup, Collapsable } from "../../../src";
import * as AvatarStories from "./Avatar.stories";
import * as AvatarGroupStories from "./AvatarGroup.stories";

<Meta of={AvatarStories} />

<Title>Avatar</Title>

<Anchor storyId="avatar">
  <div className="hidden-anchor">
    <Subheading>Avatar</Subheading>
  </div>
</Anchor>

<Description of={Avatar} />

<Primary />

<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={AvatarStories._Avatar} />
</Collapsable>

<Anchor storyId="group">
  <Subheading>Group</Subheading>
</Anchor>
<Description of={AvatarGroupStories._AvatarGroup} />
<Canvas withToolbar of={AvatarGroupStories._AvatarGroup} />
<Collapsable className="control-panel">
  <div>Controls</div>
  <Controls of={AvatarGroupStories._AvatarGroup} />
</Collapsable>

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" />

<Anchor storyId="sample">
  <Subheading>Example</Subheading>
</Anchor>

```tsx
import { Avatar, AvatarGroup } from "@react-fabric/core";

export const Example = () => {
  return (
    <AvatarGroup>
      <Avatar
        name="Avatar Name"
        avatar="Avatar image"
        fallbackIcon="Fallback svg"
      />
    </AvatarGroup>
  );
};
```
