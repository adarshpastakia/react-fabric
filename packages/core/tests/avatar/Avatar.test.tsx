/*
 * React Fabric
 * @version: 1.0.0
 *
 *
 * The MIT License (MIT)
 * Copyright (c) 2024 Adarsh Pastakia
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of this software
 * and associated documentation files (the "Software"), to deal in the Software without restriction,
 * including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all copies or substantial
 * portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
 * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

import { faker } from "@faker-js/faker";
import { composeStories } from "@storybook/react";
import { fireEvent, render } from "@testing-library/react";
import * as stories from "../../stories/components/avatar/Avatar.stories";
import { mdiAbacus } from "@mdi/js";

const { Tester } = composeStories(stories);

describe("Avatar", () => {
  it("should render avatar", () => {
    const fragment = render(
      <Tester name="Smeghead" bg="lilac" color="coral" size="md" />,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render text avatar", () => {
    const fragment = render(
      <Tester variant="text" name="Smeg head" size="2rem" />,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render fallback avatar", () => {
    const fragment = render(
      <Tester name="Smeg head" avatar="none" fallbackIcon={mdiAbacus} />,
    );
    fireEvent.error(fragment.container.querySelector("img") as HTMLElement);
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render fallback icon font", () => {
    const fragment = render(
      <Tester name="Smeg head" fallbackIcon="mdi mdi-alien" />,
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render image avatar", () => {
    const fragment = render(
      <Tester
        rounded
        variant="text"
        name="Smeg head"
        avatar="https://avatars.githubusercontent.com/u/47371134"
      />,
    );
    fireEvent.load(fragment.container.querySelector("img") as HTMLElement);
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });
});
