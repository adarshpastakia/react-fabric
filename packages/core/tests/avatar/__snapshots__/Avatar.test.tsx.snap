// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render avatar 1`] = `"<div><div class="flex gap-2 items-center p-4"><dfn role="presentation" class="fabric-avatar select-none relative overflow-hidden inline-block box-content leading-none not-italic text-center rounded" style="font-size: 1.5rem;" data-clickable="false"><svg viewBox="0 0 36 36" fill="none" role="img" xmlns="http://www.w3.org/2000/svg" width="64" height="64"><mask id=":r0:" maskUnits="userSpaceOnUse" x="0" y="0" width="36" height="36"><rect width="36" height="36" fill="#FFFFFF"></rect></mask><g mask="url(#:r0:)"><rect width="36" height="36" fill="#92A1C6"></rect><rect x="0" y="0" width="36" height="36" transform="translate(6 2) rotate(252 18 18) scale(1)" fill="#F0AB3D" rx="6"></rect><g transform="translate(4 -1) rotate(2 18 18)"><path d="M15 19c2 1 4 1 6 0" stroke="#000000" fill="none" stroke-linecap="round"></path><rect x="12" y="14" width="1.5" height="2" rx="1" stroke="none" fill="#000000"></rect><rect x="22" y="14" width="1.5" height="2" rx="1" stroke="none" fill="#000000"></rect></g></g></svg></dfn></div></div>"`;

exports[`Avatar should render fallback avatar 1`] = `"<div></div><div></div><div><div class="flex gap-2 items-center p-4"><dfn role="presentation" class="fabric-avatar select-none relative overflow-hidden inline-block box-content leading-none not-italic text-center rounded" data-clickable="false"><svg viewBox="0 0 24 24"><path fill="currentColor" d="M5 5H7V11H5V5M10 5H8V11H10V5M5 19H7V13H5V19M10 13H8V19H10V17H15V15H10V13M2 21H4V3H2V21M20 3V7H13V5H11V11H13V9H20V15H18V13H16V19H18V17H20V21H22V3H20Z"></path></svg></dfn></div></div>"`;

exports[`Avatar should render fallback icon font 1`] = `"<div></div><div></div><div></div><div><div class="flex gap-2 items-center p-4"><dfn role="presentation" class="fabric-avatar select-none relative overflow-hidden inline-block box-content leading-none not-italic text-center rounded" data-clickable="false"><i class="mdi mdi-alien"></i></dfn></div></div>"`;

exports[`Avatar should render image avatar 1`] = `"<div></div><div></div><div></div><div></div><div><div class="flex gap-2 items-center p-4"><dfn role="presentation" class="fabric-avatar select-none relative overflow-hidden inline-block box-content leading-none not-italic text-center rounded-full" data-clickable="false"><div class="contents"><img alt="https://avatars.githubusercontent.com/u/47371134" class="align-baseline" src="https://avatars.githubusercontent.com/u/47371134"></div></dfn></div></div>"`;

exports[`Avatar should render text avatar 1`] = `"<div></div><div><div class="flex gap-2 items-center p-4"><dfn role="presentation" class="fabric-avatar select-none relative overflow-hidden inline-block box-content leading-none not-italic text-center rounded" style="font-size: 2rem;" data-clickable="false"><svg role="img"><text x="50%" y="50%" dy=".1em" dominant-baseline="middle" text-anchor="middle" style="font-size: .5em; font-weight: 500;">Sh</text></svg></dfn></div></div>"`;
